File: sorting/quickSort/quick.c
#include<stdio.h>
void swap(int *arr,int i,int j){
    int t=arr[i];
    arr[i]=arr[j];
    arr[j]=t;
}
int partition(int *arr,int low,int up){
    int pivot=low+((up-low)/2);
    swap(arr,pivot,up);
    pivot=up;
    int i=low;
    int j=up-1;
    while(i<=j){
        if(arr[i]<arr[pivot]){
            i++;
        }
        else if(arr[j]>arr[pivot]){
            j--;
        }
        else if(i<=j){
            swap(arr,i,j);
            i++;
            j--;
        }
    }
    swap(arr,i,pivot);
    return i;
}
void sort(int *arr,int low,int up){
    if(low<up){
        int mid=partition(arr,low,up);
        sort(arr,low,mid-1);
        sort(arr,mid+1,up);
    }
}
int main(){
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    sort(arr,0,n-1);
    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);
    }
}

File: sorting/quickSort/quick.py
def swap(l,i,j):
    l[i],l[j]=l[j],l[i]
def partition(l,low,up):
    pivot=low+((up-low)//2)
    l[pivot],l[up]=l[up],l[pivot]
    pivot=up
    i=low
    j=up-1
    while(i<=j):
        if(l[i]<l[pivot]):
            i+=1
        elif(l[j]>l[pivot]):
            j-=1
        elif(i<j):                 # i<j, will bascially check l[i]>l[j] and greter than and less than pivot repectively 
                                   # since first and second if will check the complelement of this else if
                                   #  creiteira and since they have not passed them
                                   #they will come here
            swap(l,i,j)
            i+=1
            j-=1
    swap(l,pivot,i)
    return i
def sort(l,low,up):
    if(low<up):
        mid=partition(l,low,up)
        sort(l,low,mid-1)
        sort(l,mid+1,up)
l=[5,3,1,2,4]
sort(l,0,len(l)-1)
print(l)
